app/src/androidTest/java/com/github/se/project/ui/components/SingleSubjectSelectorTest.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.se.project.ui.components
+
+/*@RunWith(AndroidJUnit4::class)
+class SingleSubjectSelectorTest {
+
+  private var settedUp = false
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    val selectedSubject = mutableStateOf(Subject.NONE)
+    Thread.sleep(10000)
+    runOnUiThread {
+      composeTestRule.setContent { SubjectSelector(selectedSubject = selectedSubject) }
+    }
+    composeTestRule.waitForIdle()
+    settedUp = true
+  }
+
+  @Test
+  fun selectingSubjectUpdatesCorrectly() {
+    assert(settedUp)
+    composeTestRule.waitforIdle()
+
+    // Initially, dropdown items are not displayed
+    Subject.entries.forEach { subject ->
+      if (subject == Subject.NONE) return@forEach
+      composeTestRule.onNodeWithTag("dropdown${subject.name}").assertDoesNotExist()
+    }
+
+    // Click on the subject button to open the dropdown
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+
+    // **Wait for the UI to settle after the click**
+    composeTestRule.waitForIdle()
+
+    // Now, dropdown items should be displayed
+    Subject.entries.forEach { subject ->
+      if (subject == Subject.NONE) return@forEach
+      composeTestRule.onNodeWithTag("dropdown${subject.name}").assertIsDisplayed()
+    }
```

Comment:
Instead of commenting tests that bugs, we could think of putting them all in a wiki page so that we do not loose them and we do not keep a lot of commented code in our main branch.

Reply:
That's a good idea! Will do that in the future